### PR TITLE
translation features.md to Chinese

### PR DESCRIPTION
**Pull request by Dustyposa**
We have a small team doing that translation en docs to Chinese.
If you are interested on this thing. Can join us,and speed the translate.:smile:

### CODE REVIEW

File: docs/zh/docs/features.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+}
+
+my_second_user: User = User(**second_user_data)
+```
+
+
+!!! info
+    `**second_user_data` 意思是:
+
+    直接将`second_user_data`字典的键和值直接作为key-value参数传递，等同于：`User(id=4, name="Mary", joined="2018-11-30")`
```
Comment (waynerv) [inline docs/zh/docs/features.md]: duplicate "直接", suggest to be "将`second_user_data`字典的键值对直接作为关键字参数传递"

File: docs/zh/docs/features.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+### 编辑器支持
+
+所有的框架都被设计得易于使用且直观，所有的决定都在开发之前就在多个编辑器上进行了测试，来确保最佳的开发体验。
```
Comment (waynerv) [inline docs/zh/docs/features.md]: "所有的框架" should be "整个框架"

File: docs/zh/docs/features.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+在最近的 Python 开发者调查中，我们能看到<a href="https://www.jetbrains.com/research/python-developers-survey-2017/#tools-and-features" class="external-link" target="_blank">被使用最多的功能是"自动补全"</a>。
+
+整个 **FastAPI** 框架就是基于这一点的。任何地方都可以进行自动补全。
+
+你就不需要经常再来看这个文档。
```
Comment (waynerv) [inline docs/zh/docs/features.md]: “你不会需要经常回来看文档”  reads better.

File: docs/zh/docs/features.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+* <a href="https://code.visualstudio.com/" class="external-link" target="_blank">Visual Studio Code</a> 中:
+
+![editor support](https://fastapi.tiangolo.com/img/vscode-completion.png)
+
+* <a href="https://www.jetbrains.com/pycharm/" class="external-link" target="_blank">PyCharm</a> 中:
+
+![editor support](https://fastapi.tiangolo.com/img/pycharm-completion.png)
+
+你将在代码中完成你在之前可能曾认为不可能的事。例如，在来自请求 JSON 体（可能是嵌套的）中的键`price`。
```
Comment (waynerv) [inline docs/zh/docs/features.md]: "completion in code" should be "代码补全" not "在代码中完成"

File: docs/zh/docs/features.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+不会再输错键名，来回翻看文档，或者来回滚动寻找你最后使用的`username`或者`user_name`。
+
+
+
+### 总结
```
Comment (waynerv) [inline docs/zh/docs/features.md]: "Short" should be “简洁" or "简短"?

File: docs/zh/docs/features.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+
+任何类型都有合理的**默认值**，任何和地方都有可选配置。所有的参数被微调，来满足你的需求，定义成你需要的API。
+
+但是默认情况下，所有都是**“能够工作”**。
```
Comment (waynerv) [inline docs/zh/docs/features.md]: "在默认情况下，一切都能顺利工作" reads better?

File: docs/zh/docs/features.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+* 100% 测试覆盖率。
+* 代码库 100% 类型注释。
+
+## Pydantic 特性
+
+**FastAPI** 和 <a href="https://pydantic-docs.helpmanual.io" class="external-link" target="_blank"><strong>Pydantic</strong></a> 完全兼容(并基于)。所以，你有的其他的 Pydantic 代码也能正常工作。
+
+兼容包括基于 Pydantic 的外部库， 例如用与数据库的 <abbr title="对象关系映射">ORM</abbr>s, <abbr title="对象文档映射">ODM</abbr>s。
+
+这也意味着在很多情况下，你可以将从请求中获得的相同对象**直接传到数据库**，因为所有的验证都是自动的。
```
Comment (waynerv) [inline docs/zh/docs/features.md:186]: “因为所有的验证都是自动的。”should be “因为一切都会被自动验证”

File: docs/zh/docs/features.md [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+
+反之亦然，在很多情况下，你也可以将从数据库中获取的对象**直接传到客户端**。
+
+通过 **FastAPI** 你可以获得所有 **Pydantic**  (FastAPI 基于 Pydantic 做了所有的数据处理)：
```
Comment (waynerv) [inline docs/zh/docs/features.md:190]: missing “features”

通过 **FastAPI** 你可以获得所有 **Pydantic** 的特性  (因为 FastAPI 整个的数据处理部分都是基于 Pydantic 的)：

File: docs/zh/docs/features.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+
+* **更简单**：
+    * 没有新的模式定义 micro-language 需要学习。
+    * 如果你知道 Python types，你就知道如何使用 Pydantic。
```
Comment (waynerv) [inline docs/zh/docs/features.md:194]: 如果你了解 Python 的类型

File: docs/zh/docs/features.md [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+兼容包括基于 Pydantic 的外部库， 例如用与数据库的 <abbr title="对象关系映射">ORM</abbr>s, <abbr title="对象文档映射">ODM</abbr>s。
+
+这也意味着在很多情况下，你可以将从请求中获得的相同对象**直接传到数据库**，因为所有的验证都是自动的。
+
+反之亦然，在很多情况下，你也可以将从数据库中获取的对象**直接传到客户端**。
+
+通过 **FastAPI** 你可以获得所有 **Pydantic**  (FastAPI 基于 Pydantic 做了所有的数据处理)：
+
+* **更简单**：
+    * 没有新的模式定义 micro-language 需要学习。
```
Comment (waynerv) [inline docs/zh/docs/features.md:193]: 不需要学习一套新的规则或语言用来定义模式

File: docs/zh/docs/features.md [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+* 和你 **<abbr title="集成开发环境，和代码编辑器类似">IDE</abbr>/<abbr title="一个检查代码错误的程序">linter</abbr>/brain** 适配:
+    * 因为 pydantic 数据结构仅仅是你定义的类的实例；自动补全，linting，mypy 以及你的直觉应该可以和你验证的数据一起正常工作。 
+* **更快**：
+    * 在 <a href="https://pydantic-docs.helpmanual.io/#benchmarks-tag" class="external-link" target="_blank">基准测试</a> 中，Pydantic 比其他被测试的库都要快。
+* 验证**复杂结构**:
+    * 使用分层的 Pydantic 模型, Python `typing`的 `List` 和 `Dict` 等等。
+    * 验证器允许复杂的数据架构能被清晰简单地定义，检查并记录为 JSON Schema。
+    * 你可以拥有深度**嵌套的 JSON** 对象并对它们进行验证和注释。
+* **可扩展**:
+    * Pydantic 允许定义自定义数据类型或者你可以用验证器装饰器对被装饰的模型上的方法扩展验证。
```
Comment (waynerv) [inline docs/zh/docs/features.md:204]: Pydantic 允许自定义数据类型，或者你可以用验证器装饰器装饰数据模型的方法来扩展验证功能。

File: docs/zh/docs/features.md [hunk 12]
```diff
@@ -0,0 +1,205 @@
+# 特性
+
+## FastAPI 特性
+
+**FastAPI** 提供了以下内容：
+
+### 基于开放标准
+
+
+* 用于创建API的 <a href="https://github.com/OAI/OpenAPI-Specification" class="external-link" target="_blank"><strong>OpenAPI</strong></a> 包含了<abbr title="也被叫做: endpoints, routes">路径</abbr><abbr title="也叫做HTTP方法, 例如POST, GET, PUT, DELETE">操作</abbr>，请求参数，请求体，安全性等的声明。
```
Comment (RunningIkkyu) [inline docs/zh/docs/features.md:10]: Leave space before and after `API`:
"用于创建API的" should be  "用于创建 API 的"

File: docs/zh/docs/features.md [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+* 使用 <a href="http://json-schema.org/" class="external-link" target="_blank"><strong>JSON Schema</strong></a> ( OpenAPI 本身就是基于 JSON Schema 的)自动生成数据模型文档。
```
Comment (RunningIkkyu) [inline docs/zh/docs/features.md:11]: 1. 应该使用中文符号（括号）。
2. as 应该翻译出来：因为

```
* 使用 <a href="http://json-schema.org/" class="external-link" target="_blank"><strong>JSON Schema</strong></a> 自动生成数据模型文档（因为OpenAPI 本身就是基于 JSON Schema 的）。
```

File: docs/zh/docs/features.md [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+* 经过了缜密的研究后围绕这些标准而设计。并非狗尾续貂。
+* 这也允许了在很多语言中自动**生成客户端代码**。
+### 自动生成文档
```
Comment (RunningIkkyu) [inline docs/zh/docs/features.md:14]: Better leave a blank line before this line.

File: docs/zh/docs/features.md [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+
+交互式 API 文档以及具探索性 web 界面。因为该框架是基于 OpenAPI，所以有很多可选选项，默认情况下只设置了两个。
```
Comment (RunningIkkyu) [inline docs/zh/docs/features.md:16]: It's hard to understand "默认情况下只设置了两个". 补全主语宾语会更容易理解。

因为该框架是基于 OpenAPI，所以有很多可选项，**FastAPI** 默认自带两个交互式 API 文档。

File: docs/zh/docs/features.md [hunk 16]
```diff
@@ -0,0 +1,205 @@
...
+!!! info
+    `**second_user_data` 意思是:
+
+    直接将`second_user_data`字典的键和值直接作为key-value参数传递，等同于：`User(id=4, name="Mary", joined="2018-11-30")`
+
+### 编辑器支持
+
+整个框架都被设计得易于使用且直观，所有的决定都在开发之前就在多个编辑器上进行了测试，来确保最佳的开发体验。
+
+在最近的 Python 开发者调查中，我们能看到<a href="https://www.jetbrains.com/research/python-developers-survey-2017/#tools-and-features" class="external-link" target="_blank">被使用最多的功能是"自动补全"</a>。
```
Comment (RunningIkkyu) [inline docs/zh/docs/features.md:77]: Leave a space before and after `<a href="https://www.jetbrains.com/research/python-developers-survey-2017/#tools-and-features" class="external-link" target="_blank">`

File: docs/zh/docs/features.md [hunk 17]
```diff
@@ -0,0 +1,205 @@
...
+
+整个 **FastAPI** 框架就是基于这一点的。任何地方都可以进行自动补全。
+
+你不会需要经常回来看文档。
```
Comment (RunningIkkyu) [inline docs/zh/docs/features.md:81]: "不会需要"  -> "几乎不需要"

File: docs/zh/docs/features.md [hunk 18]
```diff
@@ -0,0 +1,205 @@
...
+
+在这里可以看到你的编辑器是如何帮助你的：
```
Comment (RunningIkkyu) [inline docs/zh/docs/features.md:83]: "你的编辑器可能会这样帮助你:"   可能更符合语言习惯。

File: docs/zh/docs/features.md [hunk 19]
```diff
@@ -0,0 +1,205 @@
...
+
+* <a href="https://code.visualstudio.com/" class="external-link" target="_blank">Visual Studio Code</a> 中:
+
+![editor support](https://fastapi.tiangolo.com/img/vscode-completion.png)
+
+* <a href="https://www.jetbrains.com/pycharm/" class="external-link" target="_blank">PyCharm</a> 中:
+
+![editor support](https://fastapi.tiangolo.com/img/pycharm-completion.png)
+
+你将能进行代码补全，这是在之前你可能曾认为不可能的事。例如，在来自请求 JSON 体（可能是嵌套的）中的键`price`。
```
Comment (RunningIkkyu) [inline docs/zh/docs/features.md:93]: Leave a space before `price`.

File: docs/zh/docs/features.md [hunk 20]
```diff
@@ -0,0 +1,205 @@
...
+
+不会再输错键名，来回翻看文档，或者来回滚动寻找你最后使用的`username`或者`user_name`。
```
Comment (RunningIkkyu) [inline docs/zh/docs/features.md:95]: Leave spaces before and after English words.

File: docs/zh/docs/features.md [hunk 21]
```diff
@@ -0,0 +1,205 @@
...
+
+
+
+### 简洁
+
+任何类型都有合理的**默认值**，任何和地方都有可选配置。所有的参数被微调，来满足你的需求，定义成你需要的API。
```
Comment (RunningIkkyu) [inline docs/zh/docs/features.md:101]: Leave spaces before and after English words.

File: docs/zh/docs/features.md [hunk 22]
```diff
@@ -0,0 +1,205 @@
...
+**FastAPI** 和 <a href="https://www.starlette.io/" class="external-link" target="_blank"><strong>Starlette</strong></a> 完全兼容(并基于)。所以，你有的其他的 Starlette 代码也能正常工作。`FastAPI` 实际上是 `Starlette`的一个子类。所以，如果你已经知道或者使用 Starlette，大部分的功能会以相同的方式工作。
+
+通过 **FastAPI** 你可以获得所有 **Starlette** 的特性 ( FastAPI 就像加强版的 Starlette )：
+
+* 令人惊叹的性能。它是 <a href="https://github.com/encode/starlette#performance" class="external-link" target="_blank">Python 可用的最快的框架之一，和 **NodeJS** 及 **Go** 相当</a>。
+* **支持 WebSocket** 。
+* **支持 GraphQL** 。
+* 后台任务处理。
+* Startup 和 shutdown 事件。
+* 测试客户端基于`requests`。
```
Comment (RunningIkkyu) [inline docs/zh/docs/features.md:174]: Leave spaces before and after `requests`.

File: docs/zh/docs/features.md [hunk 23]
```diff
@@ -0,0 +1,205 @@
...
+* **更简单**：
+    * 没有新的模式定义 micro-language 需要学习。
+    * 如果你知道 Python types，你就知道如何使用 Pydantic。
+* 和你 **<abbr title="集成开发环境，和代码编辑器类似">IDE</abbr>/<abbr title="一个检查代码错误的程序">linter</abbr>/brain** 适配:
+    * 因为 pydantic 数据结构仅仅是你定义的类的实例；自动补全，linting，mypy 以及你的直觉应该可以和你验证的数据一起正常工作。 
+* **更快**：
+    * 在 <a href="https://pydantic-docs.helpmanual.io/#benchmarks-tag" class="external-link" target="_blank">基准测试</a> 中，Pydantic 比其他被测试的库都要快。
+* 验证**复杂结构**:
+    * 使用分层的 Pydantic 模型, Python `typing`的 `List` 和 `Dict` 等等。
+    * 验证器允许复杂的数据架构能被清晰简单地定义，检查并记录为 JSON Schema。
```
Comment (RunningIkkyu) [inline docs/zh/docs/features.md:201]: 1. 物作主语时 `allow` 应该翻译成 **使得** ， 参考 Collins 词典。
2. defined, checked, documented 三者为并列关系。
> V-T If one thing allows another thing to happen, the first thing creates the opportunity for the second thing to happen. 

验证器使我们能够简单清楚的将复杂的数据模式定义、检查并记录为 JSON Schema。

**Review by RunningIkkyu — CHANGES_REQUESTED** [review]

**Comment by Dustyposa** [comment]
Thanks for your review.
@waynerv @RunningIkkyu 
😄 
🚀

**Review by waynerv — APPROVED** [review]

**Review by RunningIkkyu — APPROVED** [review]

**Comment by tiangolo** [comment]
Awesome! Thanks for your contribution @Dustyposa ! :coffee: :bow: :clap: 

And thanks for the reviews @waynerv and @RunningIkkyu ! :bow: :cake:

**Pull request merged**